docs/enhacement/20201211-nodepool_uniteddeployment.md

```diff
@@ -0,0 +1,374 @@
+    // +optional
+    DeploymentTemplate *DeploymentTemplateSpec `json:"deploymentTemplate,omitempty"`
+
+    // DaemonSet template
+    // +optional
+    DaemonSetTemplate *DaemonSetTemplateSpec `json:"daemonSetTemplate,omitempty"`
+}
+
+// StatefulSetTemplateSpec defines the subset template of StatefulSet.
+type StatefulSetTemplateSpec struct {
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    Spec              appsv1.StatefulSetSpec `json:"spec"`
+}
+
+// DeploymentTemplateSpec defines the subset template of AdvancedStatefulSet.
+type DeploymentTemplateSpec struct {
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    Spec              appsv1.DeploymentSpec `json:"spec"`
+}
+
+// DaemonSetTemplateSpec defines the subset template of AdvancedStatefulSet.
+type DaemonSetTemplateSpec struct {
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    Spec              appsv1.DaemonSetSpec `json:"spec"`
+}
+
+// UnitedDeploymentUpdateStrategy defines the update performance
+// when template of UnitedDeployment is changed.
+type UnitedDeploymentUpdateStrategy struct {
+    // Includes all of the parameters a statefulset update strategy needs.
+    // +optional
+    StatefulSetUpdate *StatefulSetUpdate `json:"statefulSetUpdate,omitempty"`
+
+    // Includes all of the parameters a deployment update strategy needs.
+    // +optional
+    DeploymentUpdate *DeploymentUpdate `json:"deploymentUpdate,omitempty"`
+
+    // Includes all of the parameters a daemonset update strategy needs.
+    // +optional
+    DaemonSetUpdate *DaemonSetUpdate `json:"daemonSetUpdate,omitempty"`
+}
+
+// StatefulSetUpdate is a update strategy which allows users to control the update progress
+// by providing the partition of each subset.
+type StatefulSetUpdate struct {
+    // Indicates strategy of statefulset for each subset.
+    // +optional
+    Strategies map[string]appsv1.StatefulSetUpdateStrategy `json:"strategies,omitempty"`
+}
+
+// DeploymentUpdate is a update strategy
+type DeploymentUpdate struct {
+    // Indicates strategy of deployment for each subset.
+    // +optional
+    Strategies map[string]appsv1.DeploymentStrategy `json:"strategies,omitempty"`
+}
+
+// DaemonSetUpdate is a update strategy
+type DaemonSetUpdate struct {
+    // Indicates strategys of daemonset for each subset.
+    // +optional
+    Strategies map[string]appsv1.DaemonSetUpdateStrategy `json:"strategies,omitempty"`
+}
+
+// Topology defines the spread detail of each subset under UnitedDeployment.
+// A UnitedDeployment manages multiple homogeneous workloads which are called subset.
+// Each of subsets under the UnitedDeployment is described in Topology.
+type Topology struct {
+    // Contains the details of each subset. Each element in this array represents one subset
+    // which will be provisioned and managed by UnitedDeployment.
+    // +optional
+    Subsets []Subset `json:"subsets,omitempty"`
+}
+
+// Subset defines the detail of a subset.
+type Subset struct {
```

Comment:
Suggestion to rename Subset to"division" or "partition".

Also, how should we associate a SubSet with a node pool?

Reply:
> Suggestion to rename Subset to"division" or "partition" or just "pool".
> 
> Also, how should we associate a SubSet with a node pool?
pool +1

Reply:
ok